### ISSUE TITLE
libhybris issues for Nexus 5X (bullhead)

### ISSUE THREAD

**Issue by bhush9** [Halium/projectmanagement#17]
## Works

- [x] test_gps
- [x] test_hwcomposer
- [x] test_lights
- [x] test_vibrator (see also issue #16)
- [x] test_wifi

## Doesn't work

- [ ] test_audio

```
library "/system/lib/soundfx/libspeakerbundle.so" not found
Audio device API version 0200 doesn't match platform current 3004.test_audio: test_audio.c:71: main: Assertion `audiohw->common.version == AUDIO_DEVICE_API_VERSION_CURRENT' failed.
Aborted
```

- [ ] test_camera

```
library "libcamera_compat_layer.so" not found
Segmentation fault
```

- [ ] test_input

```
library "libis_compat_layer.so" not found
Segmentation fault
```

- [ ] test_nfc

```
Starting test_nfc.
Finding NFC hardware module.
test_nfc: test_nfc.c:156: testNfc: Assertion `hwModule != NULL' failed.
Aborted
```

- [ ] test_recorder

```
Test application for video recording using the camera
Recording start with volume up button. And stops with volume down.
The result is stored to /root/test_video.avi

library "libcamera_compat_layer.so" not found
Segmentation fault
```

- [ ] test_sensors : just segfaults

## Broken test in general
- test_egl
- test_glesv2
- test_media

**Comment by bhush9** [comment]
With the commit above now test_sensors is working.

**Comment by ghost** [comment]
such a bummer... Can a new halium version fix these?

**Comment by rahmanshaber** [comment]
is anyone working on this device?any progress?

**Comment by jiri-one** [comment]
Any news guys?

**Comment by jbruechert** [comment]
Not really, it mostly works. Some of the tests are just broken but actually camera works with gst-droid.

**Comment by lapinus** [comment]
Hi! How can I compile this and try it myself? Thanks!!!!

**Comment by doniks** [comment]
docs.halium.org

**Comment by AsciiWolf** [comment]
Any news about Nexus 5X support?

**Comment by lapinus** [comment]
Hey man, my repo was literally just some experimentation with halium and
UT. The furthest I got was a fully functional (I could ssh in via USB)
version of Ubuntu but the display server just wouldn't work. It was stuck
on the Google logo.

**Comment by walidham** [comment]
I'm port UT for bullhead, i have a good progress, i got GUI, sound, vibrator, power button, touchscreen work. My work is based on halium-9.0
![bullhead](https://user-images.githubusercontent.com/6183661/192134917-23e074be-5ad2-4529-8e0c-f225976df64e.jpg)

**Comment by lapinus** [comment]
Wow! That's amazing! Congrats.

Are you planning to make a video about it or something? I don't have my 5x
anymore as it broke. But GG!

Il giorno dom 25 set 2022 alle ore 08:32 walid hammami <
***@***.***> ha scritto:

**Comment by Russanandres** [comment]
Hi. Thats cool. It can run sailfishOS 2/3/4 with hailum? I really want SFOS for our devcie.

**Comment by jvschiavo** [comment]
> I'm port UT for bullhead, i have a good progress, i got GUI, sound, vibrator, power button, touchscreen work. My work is based on halium-9.0 ![bullhead](https://user-images.githubusercontent.com/6183661/192134917-23e074be-5ad2-4529-8e0c-f225976df64e.jpg)

What is the situation now on 2026?
